src/test/java/io/camunda/testing/assertions/JobAssertTest.java

```diff
@@ -106,6 +108,65 @@ void testHasRetries() throws InterruptedException {
       assertThat(actual).hasRetries(1);
     }
 
+    @Test
+    void testHasAnyIncidents() throws InterruptedException {
+      // given
+      Utilities.deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      Utilities.startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode("error")
+          .errorMessage("error occurred")
+          .send()
+          .join();
+
+      // then
+
+      assertThat(actual).hasAnyIncidents();
+    }
+
+    @Test
+    void testHasNoIncidents() throws InterruptedException {
+      // given
+      Utilities.deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      Utilities.startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      // then
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      assertThat(actual).hasNoIncidents();
+    }
+
+    @Test
+    void testExtractLatestIncident() throws InterruptedException {
+      // given
+      Utilities.deployProcess(client, ProcessPackLoopingServiceTask.RESOURCE_NAME);
+      Utilities.startProcessInstance(client, ProcessPackLoopingServiceTask.PROCESS_ID);
+
+      // when
+      final ActivateJobsResponse jobActivationResponse = activateSingleJob();
+
+      final ActivatedJob actual = jobActivationResponse.getJobs().get(0);
+      client
+          .newThrowErrorCommand(actual.getKey())
+          .errorCode("error")
+          .errorMessage("error occurred")
+          .send()
+          .join();
+
+      final IncidentAssert incidentAssert = assertThat(actual).extractLatestIncident();
+
+      // then
+      Assertions.assertThat(incidentAssert).isNotNull();
```

Comment:
🙃 It might make more sense to assert the incident key has a sensible value (> 0). The incident assert will never be null if we manage to extract it. The value of the key could be wrong though.

Same for the extract test in `ProcessInstanceAssertTest`.